marlowe-playground-client/src/Modal/ViewHelpers.purs

```diff
@@ -1,12 +1,30 @@
 module Modal.ViewHelpers
   ( modalHeaderTitle
+  , modalHeaderWithClose
   ) where
 
-import Halogen.HTML (ClassName(ClassName), HTML, div, h2, text)
-import Halogen.HTML.Properties (classes)
+import Prelude hiding (div)
+import Data.Maybe (Maybe(..))
+import Halogen (ComponentHTML)
+import Halogen.Classes (closeModal)
+import Halogen.HTML (ClassName(ClassName), HTML, div, h2, img, text)
+import Halogen.HTML.Events (onClick)
+import Halogen.HTML.Properties (class_, classes, src)
 
 modalHeaderTitle :: forall w i. String -> HTML w i
 modalHeaderTitle title =
   div [ classes [ ClassName "modal-header" ] ]
     [ h2 [ classes [ ClassName "title" ] ] [ text title ]
     ]
+
+modalHeaderWithClose ::
+  forall m action slots.
+  String ->
+  action ->
+  ComponentHTML action slots m
```

Comment:
I can't see why this is `ComponentHTML` rather than `HTML`... 🤔

Reply:
Because of my lack of knowledge of Halogen 😅... 

Refactored to 
```purescript
modalHeaderWithClose ::
  forall w action.
  String ->
  action ->
  HTML w action
```

And now will collapse. Thanks for the tip